### PR TITLE
[bitnami/contour] feat: allow custom metrics port for service monitor and allow enabling/disabling hostport per port

### PR DESCRIPTION
**Pull request by AdamKobi**
### Description of the change

When deploying multiple Contour releases on the same cluster using `envoy.useHostPort` which is enabled by default and required when using NLB,  it automatically also configures the metrics port to be a `hostPort` which is not required and limits the ports used on this node.
Additionally the `envoy/serviceMonitor` is hard-coded to use port 8002 for metrics which means it is impossible to have both a custom port for metrics (required when using hostPort: true and another release of Contour on same cluster) and have a working serviceMonitor for this cluster.

This fixes 2 issues:
1. provide the ability to run only http/https ports as `hostPorts` while running metrics port as regular port
2. provide the ability to point service to the correct (envoy.service.ports.metrics) metrics port when using a custom one 

### Benefits

1. Use multiple releases of Contour on the same cluster without changing the metrics port (if running with `envoy.useHostPort.metrics: false`
2. Point to correct metrics port when using a custom metrics port

### Possible drawbacks

Breaking change: for users using `envoy.useHostPort: false` this can be a breaking change.
I tried to be as backward compatible as possible by providing the same  values as before for the `envoy.useHostPort.<port>` by setting everything to true which is identical to `envoy.useHostPort: true` but users which disabled this might have issues.
Not sure if this should be release 14.0.0 or just a feat i.e. release 13.2.0

### Applicable issues

Couldn't find

### Additional information

Breaking change for users using `envoy.useHostPort: false`

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by adam-oligo** [comment]
any update?

**Review by joancafom — COMMENTED** [review]
Thanks for your changes! I have left you small comment regarding the approach in order to make this backwards compatible. Please, take a look when you can :)

File: bitnami/contour/templates/envoy/daemonset.yaml [hunk 1]
```diff
@@ -162,7 +162,7 @@ spec:
           {{- end }}
           ports:
             - containerPort: {{ .Values.envoy.containerPorts.http }}
-              {{- if .Values.envoy.useHostPort }}
+              {{- if .Values.envoy.useHostPort.http }}
```
[the remark this answers is not in the document]
Comment (AdamKobi) [reply bitnami/contour/templates/envoy/daemonset.yaml]: fixed, thanks great trick! ❤️

**Review by joancafom — APPROVED** [review]
Thank you so much for your changes, LGTM! 🚀

**Comment by 3deep5me** [comment]
Does the NLB really binds to hostPort 443/80 on default?
On other cloud-providers normally the corresponding ports are opened on the host automatically if using service type: LoadBalancer. S.
```
test-contour-envoy   LoadBalancer   10.39.103.86   51.158.131.58   80:31526/TCP,443:31819/TCP   44m
```
I stumbled upon this issue when I wanted to install a second contour and I was surprised that even the metric port is published as host port by default and I wondered if this is really necessary.

**Comment by 3deep5me** [comment]
I'm a bit confused how should i disable the hostPorts if `useHostPort` will be deprecated in the future?
And would you be open for a PR to include the depreciation notes into the docs?

**Pull request merged**